crates/networking/manager/src/gossipsub/validate/beacon_block.rs

```diff
@@ -0,0 +1,164 @@
+        })
+    {
+        return Ok(ValidationResult::Ignore(
+            "Signature already received".to_string(),
+        ));
+    }
+
+    // [REJECT] The proposer signature, signed_beacon_block.signature, is valid with respect to the
+    // proposer_index pubkey.
+    if !latest_state_in_db.verify_block_signature(block)? {
+        return Ok(ValidationResult::Reject("Invalid signature".to_string()));
+    }
+
+    // [IGNORE] The block's parent (defined by block.parent_root) has been seen.
+    if let Some(parent) = store
+        .db
+        .beacon_block_provider()
+        .get(block.message.parent_root)?
+    {
+        // [REJECT] The block is from a higher slot than its parent.
+        if block.message.slot > parent.message.slot + 1 {
+            return Ok(ValidationResult::Reject(
+                "Block is from a higher slot than expected".to_string(),
+            ));
+        }
+    } else {
+        return Ok(ValidationResult::Ignore(
+            "Parent block not found".to_string(),
+        ));
+    }
+
+    let finalized_checkpoint = store.db.finalized_checkpoint_provider().get()?;
+    // [REJECT] The current finalized_checkpoint is an ancestor of block.
+    if !store.get_checkpoint_block(block.message.parent_root, finalized_checkpoint.epoch)?
+        == finalized_checkpoint.root
+    {
+        return Ok(ValidationResult::Reject(
+            "Finalized checkpoint is not an ancestor".to_string(),
+        ));
+    }
+
+    // [REJECT] The block is proposed by the expected proposer_index for the block's slot.
+    if !latest_state_in_db.get_beacon_proposer_index(Some(block.message.slot))?
+        == block.message.proposer_index
+    {
+        return Ok(ValidationResult::Reject(
+            "Proposer index is incorrect".to_string(),
+        ));
+    }
+
+    // [REJECT] The block's execution payload timestamp is correct with respect to the slot.
+    if !block.message.body.execution_payload.timestamp
+        == latest_state_in_db.compute_timestamp_at_slot(block.message.slot)
+    {
+        return Ok(ValidationResult::Reject(
+            "Execution payload timestamp is incorrect".to_string(),
+        ));
+    }
+
+    let proposer_bls_execution_change = &block
+        .message
+        .body
+        .bls_to_execution_changes
+        .get(block.message.proposer_index as usize)
+        .ok_or(anyhow!("Invalid index for signed bls to execution change"))?
+        .message;
+
+    // [IGNORE] The signed_bls_to_execution_change is the first valid signed bls to execution change
+    // received for the validator with index.
+    if cached_db
+        .cached_bls_to_execution_signature
+        .read()
+        .await
+        .contains(&AddressSlotIdentifier {
+            address: validator.public_key.clone(),
+            slot: block.message.slot,
+        })
+    {
+        return Ok(ValidationResult::Ignore(
+            "Signature already received".to_string(),
+        ));
+    }
+
```

Comment:
[IGNORE] current_epoch >= CAPELLA_FORK_EPOCH, where current_epoch is defined by the current wall-clock time.
[IGNORE] The signed_bls_to_execution_change is the first valid signed bls to execution change received for the validator with index signed_bls_to_execution_change.message.validator_index.
[REJECT] All of the conditions within process_bls_to_execution_change pass validation.


aren't we missing the last case here?